### PR TITLE
Fix ingress test for private CAPA clusters

### PR DESCRIPTION
**Pull request by whites11**
### What this PR does

Towards: https://github.com/giantswarm/roadmap/issues/3099

The ingress test for private capa clusters didn't work.
I needed to fix a few things here and there in order to make it run.
This PR is part of those fixes.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by calvix** [comment]
any more info? :p why is the change needed now or  what part of issue is it?

File: internal/common/hello.go [hunk 1]
```diff
@@ -162,25 +171,44 @@ func runHelloWorld(externalDnsSupported bool) {
...
-					},
+						},
+					}
+					return dialer.DialContext(ctx, network, addr)
 				},
 			}
 
+			if os.Getenv("HTTP_PROXY") != "" {
+				logger.Log("Detected need to use PROXY as HTTP_PROXY env var was set to %s", os.Getenv("HTTP_PROXY"))
+				transport.Proxy = http.ProxyFromEnvironment
```
Comment (AverageMarcus) [inline internal/common/hello.go:205]: Oh this is nice!

**Review by AverageMarcus — APPROVED** [review]
Great work! Thanks!

**Pull request merged**